src/COLLADA2GLTFWriter.cpp

```diff
@@ -1620,6 +1620,33 @@ bool COLLADA2GLTF::Writer::writeController(const COLLADAFW::Controller* controll
 		std::tie(type, joints, weights) = _skinData[skinControllerDataId];
 		int numberOfComponents = GLTF::Accessor::getNumberOfComponents(type);
 
+		float maxweight = 0.0, minweight = 0.0;
+		if (weights.size() > 0) {
```

Comment:
I think this is redundant with the for loop below it and can probably be remove unless I'm missing something.

Reply:
In my opinion, if this condition is not imposed, if in some case, weights.size() = 0, we will still end up trying to access 1st element of weights vector. But since there is no element in the weights vector, it would result in segmentation fault. A solution to this problem is to assign constant values to maxweight and minweight initially which are equal to the upper and lower bound of float datatype. But the current solution seemed to be more robust to me.